rdt/transformers/_validations.py

```diff
@@ -0,0 +1,160 @@
+"""Validations for multi-column transformers."""
+import importlib
+
+from rdt.errors import TransformerInputError
+
+
+class BaseValidator:
+    """Base validation class.
+
+    The validation classes ensure that the input data is compatible with the transformers
+    and that they can be imported.
+    """
+
+    SUPPORTED_SDTYPES = []
+    VALIDATION_TYPE = None
+
+    @classmethod
+    def _validate_supported_sdtypes(cls, columns_to_sdtypes):
+        message = ''
+        for column, sdtype in columns_to_sdtypes.items():
+            if sdtype not in cls.SUPPORTED_SDTYPES:
+                message += f"Column '{column}' has an unsupported sdtype '{sdtype}'.\n"
+
+        if message:
+            message += (
+                f'Please provide a column that is compatible with {cls.VALIDATION_TYPE} data.'
+            )
+            raise TransformerInputError(message)
+
+    @classmethod
+    def validate_sdtypes(cls, columns_to_sdtypes):
+        """Validate the columns to sdtypes mapping.
+
+        This method aims to call all other sdtype validation method in the class.
+
+        Args:
+            columns_to_sdtypes (dict):
+                Mapping of column names to sdtypes.
+        """
+        for attr_name in dir(cls):
+            attr = getattr(cls, attr_name)
+            if callable(attr) and attr_name.startswith('_validate_'):
+                attr(columns_to_sdtypes)
```

Comment:
this is a little tricky. I think it's more transparent to just have the child class implement `validate_sdtypes` and call what it needs to

Reply:
Alright, done in [aac20d5](https://github.com/sdv-dev/RDT/pull/779/commits/aac20d553fdcf5c03ae4dc60d1fe37fbfee0be96)